File: C++/Test/Test_1.cpp
#include<iostream>
using namespace std;
class node{
    public:
        int num;
        node *next;
        node(int numIn);
};
class linkList{
    public:
        node *head;
        node *nCurr;
        node *nPrev;
        node *nNext;
        node *nullNode;
        linkList();
        void addFirst(int newNum);
        void showList(node *currNode);
        void makeRev();
};
node::node(int numIn){
    num=numIn;
    next=NULL;
}
linkList::linkList(){
    head=NULL;
    nCurr=NULL;
    nPrev=NULL;
    nNext=NULL;
    nullNode=NULL;
}
void linkList::addFirst(int newNum){
    node *newNode=new node(newNum);
    if(head!=NULL){
        newNode->next=head;
    }
    head=newNode;
}
void linkList::showList(node *currNode){
    if(currNode!=NULL){
        cout<<currNode->num<<" ";
        showList(currNode->next);
    }
    else{
        cout<<"End."<<endl;
    }
}
void linkList::makeRev(){
    nCurr=head;
    nNext=head->next;
    while(nNext!=NULL){
        nCurr->next=nPrev;
        nPrev=nCurr;
        nCurr=nNext;
        if(nNext->next!=NULL){
            nNext=nNext->next;
        }
        else{
            nCurr->next=nPrev;
            head=nCurr;
            break;
        }
    }
}
//**************************************
int main(){
    linkList *list=new linkList();
    for(int i=0;i<=5;i++){
        list->addFirst(i);
    }
    list->showList(list->head);
    list->makeRev();
    list->showList(list->head);
}
//**************************************

File: Test/Python/Test1.py
class node:
    def __init__(self, numIn):
        self.num=numIn
        self.next=None

class linkList:
    def __init__(self):
        self.head=None
    
    def addFirst(self, newNum):
        newNode=node(newNum)
        if self.head is not None:
            newNode.next=self.head
        else:
            self.head=newNode
        self.head=newNode
    
    def showLL(self):
        currNode=self.head
        while currNode is not None:
            print(currNode.num, end=" ")
            currNode=currNode.next
        print("")
        
    def deleteNum(self, targetNum):
        currNode=self.head
        if targetNum is self.head.num:
            self.head=self.head.next
            currNode.next=None
            return 
        while currNode is not None:
            if currNode.next.num is targetNum:
                currNode.next=currNode.next.next;
                break
            else:
                currNode=currNode.next
            
ll=linkList()
ll.addFirst(1)
ll.addFirst(2)
ll.addFirst(3)
ll.addFirst(4)
ll.showLL()
ll.deleteNum(4)
ll.showLL()